exla/lib/exla/mlir/module.ex

```diff
@@ -71,6 +71,10 @@ defmodule EXLA.MLIR.Module do
     * `:use_spmd` - enables Single-Program Multiple-Data partioning.
       This is set to true if `:num_partitions` is more than one, otherwise is `false`.
 
+    * `:module_compilation` - either `:to_mlir` or `:to_pjrt`. The default is `:to_pjrt`.
```

Comment:
@josevalim not sure about the option naming here

Reply:
It is private, so it is fine!

Reply:
If we return `%EXLA.MLIR.Module{}` in the other function, shouldn't the whole module be public?

Reply:
Good catch. We should return the string only, not the module struct.

Reply:
The public-facing function is now returning the binary directly